api/v1/dogu_types.go

```diff
@@ -208,12 +228,55 @@ func (d *Dogu) Update(ctx context.Context, client client.Client) error {
 	return nil
 }
 
+// ChangeRequeuePhase changes the requeue phase of this dogu resource and applies it to the cluster state.
+func (d *Dogu) ChangeRequeuePhase(ctx context.Context, client client.Client, phase string) error {
+	d.Status.RequeuePhase = phase
+	return d.Update(ctx, client)
+}
+
+// ChangeRequeuePhaseWithRetry refreshes the dogu resource and tries to set the requeue phase. If a conflict error occurs
+// this method will retry the operation.
+func (d *Dogu) ChangeRequeuePhaseWithRetry(ctx context.Context, client client.Client, phase string) error {
+	return retry.OnConflict(func() error {
+		err := d.refreshDoguValue(ctx, client)
+		if err != nil {
+			return err
+		}
+
+		return d.ChangeRequeuePhase(ctx, client, phase)
+	})
+}
+
+func (d *Dogu) refreshDoguValue(ctx context.Context, client client.Client) error {
+	dogu := &Dogu{}
+	err := client.Get(ctx, d.GetObjectKey(), dogu)
+	if err != nil {
+		return err
+	}
+	*d = *dogu
+
+	return nil
+}
+
 // ChangeState changes the state of this dogu resource and applies it to the cluster state.
 func (d *Dogu) ChangeState(ctx context.Context, client client.Client, newStatus string) error {
```

Comment:
Do we still need the old function? isn't it a potential source of error that shouldn't exist?